client/state/editor/test/reducer.js

```diff
@@ -0,0 +1,159 @@
+/**
+ * External dependencies
+ */
+import { expect } from 'chai';
+import deepFreeze from 'deep-freeze';
+
+/**
+ * Internal dependencies
+ */
+import {
+	EDITOR_POST_EDIT,
+	SERIALIZE,
+	DESERIALIZE
+} from 'state/action-types';
+import { posts } from '../reducer';
+
+describe( 'reducer', () => {
+	describe( '#posts()', () => {
+		it( 'should default to an empty object', () => {
+			const state = posts( undefined, {} );
+
+			expect( state ).to.eql( {} );
+		} );
+
+		it( 'should track new post draft revisions by site ID', () => {
+			const state = posts( undefined, {
+				type: EDITOR_POST_EDIT,
+				siteId: 2916284,
+				post: { title: 'Ribs & Chicken' }
+			} );
+
+			expect( state ).to.eql( {
+				2916284: {
+					'': {
```

Comment:
>Is this what we call a dirty post currently? If we want to have more than one of these, we'll likely need to generate local ids.

The empty string designates a new unsaved post draft. With this model, it's possible to have multiple simultaneous edits for any number of existing posts, but only a single new draft. I'd agree once we get to the point of locally saving the draft, the entity will be moved to `state.posts.items` with a temporary ID until the persisted ID is made available. For unsaved posts however, it seems logical to only allow a single draft at a time.